git/client.go

```diff
@@ -48,13 +48,13 @@ type RepositoryWriter interface {
 	// SwitchBranch switches from the current branch of the repository to the
 	// provided branch. If the branch doesn't exist, it is created.
 	SwitchBranch(ctx context.Context, branch string) error
-	// WriteFile creates a new file at the given path, reads from the reader and
-	// writes the content to the file. If the file already exists, its overwritten.
-	WriteFile(path string, reader io.Reader) error
-	// Commit commits any changes made to the repository.
-	// signer is an optional argument which can be provided to sign the commit using
-	// OpenPGP.
-	Commit(info Commit, signer *openpgp.Entity) (string, error)
+	// Commit writes files (if desired) and then commits any changes made to the
+	// repository.
+	// signer is an optional argument which can be provided to sign the commit
+	// using OpenPGP.
+	// files is an optional argument with file names as its key and the content
+	// of the file as the value. If the file already exists, its overwritten.
+	Commit(info Commit, signer *openpgp.Entity, files map[string]io.Reader) (string, error)
```

Comment:
If the signer and files are optional, how about creating an option type for them? Then there won't be any `nil, nil` being passed in a lot of the code that uses or tests this.

Reply:
Do you mean creating something like `CommitOptions`? I like the idea of avoiding `nil, nil` being passed in. But, would something like` files` naturally fit into a type named `CommitOptions`?

Reply:
yes. `Files` can just be an attribute of `CommitOptions` in the sense these are the files that should be part of the commit.

Reply:
+1 on using `CommitOptions` instead of `signer` and `files`.